osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,712 @@
+  registers << " segfs:0x" << std::hex << threadContext->SegFs;
+  registers << " seggs:0x" << std::hex << threadContext->SegGs;
+  registers << " segss:0x" << std::hex << threadContext->SegSs;
+  registers << " eflags:0x" << std::hex << threadContext->EFlags;
+  r["registers"] = registers.str();
+
+  return;
+}
+
+void processDumpMiscInfoStream(Row& r,
+                               MINIDUMP_DIRECTORY* dumpStreamDir,
+                               void* dumpStream,
+                               unsigned long dumpStreamSize) {
+  auto miscInfoStream = static_cast<MINIDUMP_MISC_INFO*>(dumpStream);
+
+  // Log PID, if it exists
+  if (miscInfoStream->Flags1 & MINIDUMP_MISC1_PROCESS_ID) {
+    r["pid"] = BIGINT(miscInfoStream->ProcessId);
+  }
+
+  // Log process times, if they exist
+  if (miscInfoStream->Flags1 & MINIDUMP_MISC1_PROCESS_TIMES) {
+    time_t procTimestamp = miscInfoStream->ProcessCreateTime;
+    struct tm gmt;
+    char timeBuff[64];
+    gmtime_s(&gmt, &procTimestamp);
+    strftime(timeBuff, sizeof(timeBuff), "%Y-%m-%d %H:%M:%S UTC", &gmt);
+    r["process_create_time"] = timeBuff;
+  }
+
+  return;
+}
+
+void processDumpSystemInfoStream(Row& r,
+                                 MINIDUMP_DIRECTORY* dumpStreamDir,
+                                 void* dumpStream,
+                                 unsigned long dumpStreamSize) {
+  auto systemInfoStream = static_cast<MINIDUMP_SYSTEM_INFO*>(dumpStream);
+
+  // Log system version information
+  r["major_version"] = INTEGER(systemInfoStream->MajorVersion);
+  r["minor_version"] = INTEGER(systemInfoStream->MinorVersion);
+  r["build_number"] = INTEGER(systemInfoStream->BuildNumber);
+
+  return;
+}
+
+void processDumpModuleListStream(Row& r,
+                                 MINIDUMP_DIRECTORY* dumpStreamDir,
+                                 void* dumpStream,
+                                 unsigned long dumpStreamSize,
+                                 void* dumpBase) {
+  auto moduleListStream = static_cast<MINIDUMP_MODULE_LIST*>(dumpStream);
+
+  // Log PE path
+  MINIDUMP_MODULE exeModule = moduleListStream->Modules[0];
+  auto exePath = reinterpret_cast<MINIDUMP_STRING*>(
+      static_cast<unsigned char*>(dumpBase) + exeModule.ModuleNameRva);
+  r["path"] = wstringToString(exePath->Buffer);
+
+  // Log PE version
+  VS_FIXEDFILEINFO versionInfo = exeModule.VersionInfo;
+  std::ostringstream versionStr;
+  versionStr << ((versionInfo.dwFileVersionMS >> 16) & 0xffff) << "."
+             << ((versionInfo.dwFileVersionMS >> 0) & 0xffff) << "."
+             << ((versionInfo.dwFileVersionLS >> 16) & 0xffff) << "."
+             << ((versionInfo.dwFileVersionLS >> 0) & 0xffff);
+  r["version"] = versionStr.str();
+
+  // Read exception address from the row
+  std::istringstream exAddrStr(r["exception_address"]);
+  unsigned long long exAddr;
+  exAddrStr >> std::hex >> exAddr;
+
+  // Log module that caused the exception, if any
+  if (!exAddrStr.fail()) {
```

Comment:
I think if you flip your logic here you could bail early and save yourself a level of scope:
```
if(exAddrStr.fail()){
  return;
}
```